### PR TITLE
feat: Add JSON Schema to Test destination plugin

### PR DESCRIPTION
**Pull request by disq**
Implements https://github.com/cloudquery/cloudquery/issues/16352

### CODE REVIEW

File: plugins/destination/test/client/schema.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "https://github.com/cloudquery/cloudquery/plugins/destination/test/client/spec",
+  "$ref": "#/$defs/Spec",
+  "$defs": {
+    "Spec": {
+      "properties": {
+        "error_on_write": {
+          "type": "boolean",
+          "description": "If true, will return an error on write rather than consume from the channel"
```
Comment (candiduslynx) [inline plugins/destination/test/client/schema.json:10]: you can add default here via annotating the struct:
```go
type Spec struct {
	// If true, will return an error on write rather than consume from the channel
	ErrorOnWrite bool `json:"error_on_write,omitempty" jsonschema:"default=false"`
}
```

File: plugins/destination/test/client/spec_test.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+func TestJSONSchema(t *testing.T) {
+	jsonschema.TestJSONSchema(t, JSONSchema, []jsonschema.TestCase{
+		{
+			Name: "empty spec",
+			Spec: `{}`,
+		},
+		{
+			Name: "spec with err_on_write",
+			Spec: `{"error_on_write": true}`,
```
Comment (candiduslynx) [inline plugins/destination/test/client/spec_test.go:17]: add test for `{"error_on_write":null}` & `{"error_on_write":123}` as well

File: plugins/destination/test/go.mod [hunk 3]
```diff
@@ -4,10 +4,15 @@ go 1.21.4
 
 require (
 	github.com/apache/arrow/go/v15 v15.0.0-20240114144300-7e703aae55c1
+	github.com/cloudquery/codegen v0.3.12
 	github.com/cloudquery/plugin-sdk/v4 v4.28.0
+	github.com/invopop/jsonschema v0.12.0
 	github.com/rs/zerolog v1.31.0
 )
 
+// github.com/cloudquery/jsonschema @ cqmain
```
Comment (candiduslynx) [inline plugins/destination/test/go.mod:13]: better move the replace to the end of the mod file (otherwise it might produce severla additiona sections & not keep the direct & indirect imports separated

**Review by candiduslynx — APPROVED** [review]

**Comment by candiduslynx** [comment]
@disq could you also add make gen & check for diff steps to the workflow?
